### ISSUE TITLE
Adding a minimal inheriting config to a subfolder shadows out files

### ISSUE THREAD

**Issue by leoarnold** [rubocop/rubocop#11397]
Given the configuration

```yaml
# .rubocop.yml
require:
  - rubocop-rails

AllCops:
  NewCops: enable
  DisabledByDefault: true

Rails/Output:
  Enabled: true
```

and the two Ruby files

```ruby
# lib/tool.rb
puts 'Hello'
```

and

```ruby
# lib/acme/thing.rb
puts 'Hello'
```

when I run `rubocop` I get the expected result

```
$ rubocop 
Inspecting 3 files
.CC

Offenses:

lib/acme/thing.rb:1:1: C: [Correctable] Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.
puts 'Hello'
^^^^
lib/tool.rb:1:1: C: [Correctable] Rails/Output: Do not write to stdout. Use Rails's logger if you want to log.
puts 'Hello'
^^^^

3 files inspected, 2 offenses detected, 2 offenses autocorrectable
```

Let's add another file

```yaml
# lib/.rubocop.yml 
inherit_from: '../.rubocop.yml'

Metrics/BlockLength:
  Enabled: true
```

## Expected behavior

Nothing changes, same offenses.

## Actual bahavior

The offenses are suddenly gone:

```
$ rubocop 
Inspecting 3 files
...

3 files inspected, no offenses detected
```

and - even weirder - when we add an `Include` in the base config

```yaml
# .rubocop.yml
require:
  - rubocop-rails

AllCops:
  NewCops: enable
  DisabledByDefault: true

Rails/Output:
  Enabled: true
  Include:
    - lib/**/*.rb
```

it works again as expected.

## Observations

- This will not happen with native cops (e.g. `Layout/LineLength`), so the `require` seems to play a role here.
- Originally, the absolute paths of the offending files are compared against the absolute path of the base config. After adding the inheriting config, they are now take relative from the inheriting config. If one replaces this line
  https://github.com/rubocop/rubocop/blob/f9248c8c0a61611be2e8fc59e1dcf299ff712af3/lib/rubocop/config.rb#L243
  with `Dir.pwd`, it will work again.

## RuboCop version

```
$ rubocop -V
1.42.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.1.3) [x86_64-linux]
  - rubocop-rails 2.17.4
```

**Comment by JochenLutz** [comment]
The problem still exists with Rubocop 1.79.2

**Comment by JochenLutz** [comment]
The problem still exists with Rubocop 1.81.7.

*Note to myself:* add an offense to the code before testing. 😄

**Comment by leoarnold** [comment]
I tested it again and it turned out **not** to be a problem with `rubocop`, but `rubocop-rails`. Using `git bisect` I found that [this commit](https://github.com/rubocop/rubocop-rails/commit/1126d68497e61c29e165d38f210606a71373a475) fixed the issue. Turns out that the default configuration of `rubocop-rails` was to blame for too greedy glob patterns.

**Comment by JochenLutz** [comment]
I had this issue not with `rubocop-rails`, but with some self-developed, project-specific cops.

Thanks to the linked issue, I was able to quickly identify the necessary update to our cops's `#initialize`:

```diff
-  cop_config['Include'] ||= [app/abilities/**/*.rb']
+  cop_config['Include'] ||= ['**/app/abilities/**/*.rb']
```
